### DISCUSSION TITLE
Filament Select Field with Searchable Options: Show User Input as Selectable Option at the Top

### DISCUSSION THREAD

**Discussion by heilers** [filamentphp/filament#14635] [Help]
### Package

Form builder

### Package Version

3.2.119

### How can we help you?

Hi everyone,

I’d like to implement a Select field in Filament that’s searchable through a set of model options. However, I want the user’s current input to always appear as a selectable option at the top of the dropdown, even if it doesn’t already exist in the list of options.

It’s worth mentioning that this is not a relationship field; the search only needs to run through existing options in the model.

Has anyone implemented something similar, or does anyone have suggestions on how to achieve this?

Thanks in advance for any insights you can share!

Best regards,
Hendrik

**Comment by dissto** [comment]
Maybe a [datalist](https://filamentphp.com/docs/3.x/forms/fields/text-input#autocompleting-text-with-a-datalist) could work in your case? 🤔

**Reply by heilers** [reply]
Unfortunately, there are too many values in the list. I would like to load them using the Select field method getSearchResultsUsing().

But I found a workaround with getSearchResultsUsing() and adding the current search to the result. Thanks!
